superset-frontend/cypress-base/cypress/integration/sqllab/query.test.ts

```diff
@@ -157,4 +147,52 @@ describe('SqlLab query panel', () => {
       assertSQLLabResultsAreEqual(initialResultsTable, savedQueryResultsTable);
     });
   });
+
+  it('Create a chart from a query', () => {
+    cy.intercept('/superset/sql_json/').as('queryFinished');
+    cy.intercept('**/api/v1/explore/**').as('explore');
+    cy.intercept('**/api/v1/chart/**').as('chart');
+
+    // cypress doesn't handle opening a new tab, override window.open to open in the same tab
+    cy.window().then(win => {
+      cy.stub(win, 'open', url => {
+        // eslint-disable-next-line no-param-reassign
+        win.location.href = url;
+      });
+    });
+
+    const query = 'SELECT gender, name FROM main.birth_names';
+
+    cy.get('.ace_text-input')
+      .focus()
+      .clear({ force: true })
+      .type(`{selectall}{backspace}${query}`, { force: true });
+    cy.get('.sql-toolbar button').contains('Run').click();
+    cy.wait('@queryFinished');
+
+    cy.get(
+      '.SouthPane .ant-tabs-content > .ant-tabs-tabpane-active > div button:first',
+      { timeout: 10000 },
```

Comment:
Just out of curiosity, what's the reason of this high timeout?

Reply:
Sometimes there was a small delay in rendering the component with buttons. Even though we wait for the query to finish, the test was flaky because cypress couldn't find the button because of that delay. A smaller timeout would probably work just as well though